### PR TITLE
init baroclinic gpu

### PR DESCRIPTION
**Pull request by rheacangeo**
On GPU the initilized state was actually all 0s, as the effort to keep the code from crashing on GPU actually entirely separated the gpu arrays from the cpu arrays. It is possible to pass cupy arrays (the state.quantity.data instead of the state.storage.data, which is a memory view and cannot be sliced) in and still do it this way, but this requires modifying almost all the lines of math to be cupy friendly (e.g. cp.asarray wrappers around many statements). Instead, we compute on the cpu in numpy, and from that create a DycoreState. 
This means we won't use the quantity.view feature to slice unless we revisit using a a DycoreState initialized with 0's to start with

### CODE REVIEW

**Comment by rheacangeo** [comment]
launch jenkins

File: fv3core/fv3core/initialization/baroclinic.py [hunk 1]
```diff
@@ -423,63 +425,65 @@ def init_baroclinic_state(
     Create a DycoreState object with quantities initialized to the Jablonowski &
     Williamson baroclinic test case perturbation applied to the cubed sphere grid.
     """
-
-    state = DycoreState.init_empty(metric_terms.quantity_factory)
-    nx, ny, nz = local_compute_size(state.delp.data.shape)
+    sample_quantity = metric_terms.lat
```
Comment (twicki) [inline fv3core/fv3core/initialization/baroclinic.py:428]: can we extract the initialization of the numpy state out in it's own function?

**Review by twicki — CHANGES_REQUESTED** [review]

File: fv3core/fv3core/initialization/baroclinic.py (hunk 1 above)
Comment (rheacangeo) [reply fv3core/fv3core/initialization/baroclinic.py:428]: yes! good idea

**Review by twicki — APPROVED** [review]
looks great

**Pull request merged**